<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>ATACseq or ChIPseq peak enrichment using a sliding distance • peakdistanceenrichment</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="ATACseq or ChIPseq peak enrichment using a sliding distance">
<meta property="og:description" content="Test ATACseq or ChIPseq genome regions (peaks) by
   hypergeometric enrichment with a set of genome features, 
   iterated across a sliding range of distances.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">peakdistanceenrichment</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="http://github.com/jmw86069/peakdistanceenrichment/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="peakdistanceenrichment" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#peakdistanceenrichment" class="anchor"></a>peakdistanceenrichment</h1></div>
<p>ATACseq peak enrichment using sliding scale distance.</p>
<p>The goal of the R package <code>peakdistanceenrichment</code> is to provide convenient way to reproduce the enrichment method used by Langer et al 2019, eLife.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/remotes/man/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"jmw86069/peakdistanceenrichment"</span><span class="op">)</span></pre></div>
</div>
<div id="package-reference" class="section level2">
<h2 class="hasAnchor">
<a href="#package-reference" class="anchor"></a>Package Reference</h2>
<p>A full online function reference is available via the pkgdown documentation:</p>
<p><a href="https://jmw86069.github.io/peakdistanceenrichment">Full command reference</a></p>
</div>
<div id="the-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#the-workflow" class="anchor"></a>The workflow</h2>
<p>In short:</p>
<ul>
<li>
<p>A set of “features” are provided, where each feature has a “category”.</p>
<ul>
<li>In this case, the features are RNAseq genes, and the category values are <code>-1</code> for down-regulated genes, <code>1</code> for up-regulated genes, and <code>0</code> for genes not changed.</li>
</ul>
</li>
<li>
<p>One or more sets of “peaks” are provided.</p>
<ul>
<li>In this case, there are two sets of peaks:</li>
<li>
<strong>higher accessibility</strong> ATACseq peaks</li>
<li>
<strong>lower accessibility</strong> ATACseq peaks</li>
</ul>
</li>
<li><p>A range of distances is provided, by default <code>0</code> to <code>5000</code> in kilobases, so the largest range is 5,000,000 bases.</p></li>
<li>
<p>For each set of peaks, each category is tested for enrichment against the other categories.</p>
<ul>
<li>For example features with category <code>-1</code> are tested to see if they overlap <strong>lower accessibility</strong> peaks more than you would randomly expect compared to features with category <code>0, 1</code>.</li>
<li>The features are extended across the range of distances, and this test is repeated for each distance.</li>
</ul>
</li>
</ul>
</div>
<div id="example-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#example-analysis" class="anchor"></a>Example analysis</h2>
<p>The package contains sample files sufficient to reproduce Figure 2B of the Langer et al 2019 eLife paper.</p>
<p>The steps below import the example files and create the required <code>GRanges</code> objects.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://github.com/jmw86069/peakdistanceenrichment">peakdistanceenrichment</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">GenomicRanges</span><span class="op">)</span>

<span class="co"># import hg19 chromosome sizes</span>
<span class="va">hg19file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,
   <span class="st">"hg19.sizes.txt"</span>,
   package<span class="op">=</span><span class="st">"peakdistanceenrichment"</span><span class="op">)</span>;
<span class="va">hg19df</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="va">hg19file</span>, data.table<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>;
<span class="va">hg19seqinfo</span> <span class="op">&lt;-</span> <span class="fu">Seqinfo</span><span class="op">(</span>seqnames<span class="op">=</span><span class="va">hg19df</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,
   seqlengths<span class="op">=</span><span class="va">hg19df</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>,
   isCircular<span class="op">=</span><span class="va">hg19df</span><span class="op">[</span>,<span class="fl">3</span><span class="op">]</span>,
   genome<span class="op">=</span><span class="va">hg19df</span><span class="op">[</span>,<span class="fl">4</span><span class="op">]</span><span class="op">)</span>

<span class="co">## Import features, TSS sites</span>
<span class="va">tssfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,
   <span class="st">"BAF47KD_steadystate_RNAseq_hits.bed"</span>,
   package<span class="op">=</span><span class="st">"peakdistanceenrichment"</span><span class="op">)</span>;
<span class="va">tssdf</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="va">tssfile</span>, data.table<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">tssdf</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"seqnames"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"gene"</span>, <span class="st">"hit"</span><span class="op">)</span>
<span class="va">tssgr</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="va">tssdf</span>, <span class="st">"GRanges"</span><span class="op">)</span>
<span class="co">## assign seqinfo</span>
<span class="fu">seqinfo</span><span class="op">(</span><span class="va">tssgr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">hg19seqinfo</span>;

<span class="co">## Create tss list</span>
<span class="va">tssL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>;
<span class="va">tssL</span><span class="op">$</span><span class="va">steadystate</span> <span class="op">&lt;-</span> <span class="va">tssgr</span>;


<span class="co">## Load peak files</span>
<span class="va">higherbed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,
   <span class="st">"BAF47KD_steadystate_higher_accessibility_peaks.bed"</span>,
   package<span class="op">=</span><span class="st">"peakdistanceenrichment"</span><span class="op">)</span>;
<span class="va">higherdf</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="va">higherbed</span>, data.table<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">higherdf</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"seqnames"</span>, <span class="st">"start"</span>, <span class="st">"end"</span><span class="op">)</span>;
<span class="va">highergr</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="va">higherdf</span>, <span class="st">"GRanges"</span><span class="op">)</span>
<span class="fu">seqinfo</span><span class="op">(</span><span class="va">highergr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">hg19seqinfo</span>;

<span class="va">lowerbed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,
   <span class="st">"BAF47KD_steadystate_lower_accessibility_peaks.bed"</span>,
   package<span class="op">=</span><span class="st">"peakdistanceenrichment"</span><span class="op">)</span>;
<span class="va">lowerdf</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="va">lowerbed</span>, data.table<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">lowerdf</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"seqnames"</span>, <span class="st">"start"</span>, <span class="st">"end"</span><span class="op">)</span>;
<span class="va">lowergr</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="va">lowerdf</span>, <span class="st">"GRanges"</span><span class="op">)</span>
<span class="fu">seqinfo</span><span class="op">(</span><span class="va">lowergr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">hg19seqinfo</span>;

<span class="co">## Create peak list for each set of TSS sites</span>
<span class="va">peaksL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span>;
<span class="va">peaksL</span><span class="op">$</span><span class="va">steadystate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
   higher<span class="op">=</span><span class="va">highergr</span>,
   lower<span class="op">=</span><span class="va">lowergr</span><span class="op">)</span>;</pre></div>
<p>Next, the features and peaks are tabulated across the range of distances, for each category.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="co">## Call peak_feature_ranged_counter()</span>
<span class="co">## - count features per peak</span>
<span class="co">## - using an expanded peak size</span>
<span class="va">range_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/peak_feature_ranged_counter.html">peak_feature_ranged_counter</a></span><span class="op">(</span>
   peaksL<span class="op">=</span><span class="va">peaksL</span>,
   featuresL<span class="op">=</span><span class="va">tssL</span>,
   expand_range_kb<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">2000</span><span class="op">)</span>,
   verbose<span class="op">=</span><span class="cn">FALSE</span>,
   category_colname<span class="op">=</span><span class="st">"hit"</span><span class="op">)</span>;</pre></div>
<p>The counts are tested with hypergeometric enrichment, to test each category, for each set of peaks, and at each distance. The output is a <code>data.frame</code> that include the hypergeometric P-value in column <code>"phyper.p"</code>. It also includes the values passed to <code><a href="https://rdrr.io/r/stats/Hypergeometric.html">phyper()</a></code>, specific the <code>q, m, n, k</code> values.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="va">paramsDF</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/peak_feature_ranged_enrichment.html">peak_feature_ranged_enrichment</a></span><span class="op">(</span><span class="va">range_counts</span>,
   test_cats<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"1"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"-1"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">paramsDF</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;      Experiment PeakSet Range q_drawn1 m_drawn1_undrawn1 n_drawn0_undrawn0</span>
<span class="co">#&gt; 1.1 steadystate  higher     5       60              1768             15694</span>
<span class="co">#&gt; 1.2 steadystate  higher    10       79              1768             15694</span>
<span class="co">#&gt; 1.3 steadystate  higher    15      104              1768             15694</span>
<span class="co">#&gt; 1.4 steadystate  higher    20      132              1768             15694</span>
<span class="co">#&gt; 1.5 steadystate  higher    25      157              1768             15694</span>
<span class="co">#&gt; 1.6 steadystate  higher    30      177              1768             15694</span>
<span class="co">#&gt;     k_drawn0_drawn1 test_cat phyper.p</span>
<span class="co">#&gt; 1.1             328        1 1.99e-06</span>
<span class="co">#&gt; 1.2             537        1 2.65e-04</span>
<span class="co">#&gt; 1.3             737        1 1.84e-04</span>
<span class="co">#&gt; 1.4             955        1 7.64e-05</span>
<span class="co">#&gt; 1.5            1167        1 7.10e-05</span>
<span class="co">#&gt; 1.6            1336        1 5.98e-05</span></pre></div>
<p>This result can be plotted manually, or by using the helper function <code><a href="reference/plot_ranged_enrichment.html">plot_ranged_enrichment()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="co">## Plot the enrichment P-values</span>
<span class="va">gg_pre</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/plot_ranged_enrichment.html">plot_ranged_enrichment</a></span><span class="op">(</span><span class="va">paramsDF</span>,
   test_cat_title<span class="op">=</span><span class="st">"RNAseq\ndirection"</span>,
   draw_points<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
   <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">gg_pre</span><span class="op">)</span></pre></div>
<p><img src="reference/figures/README-plot_enrichment-1.png"><!-- --></p>
<p>If you want to update the category labels, you can do something like this:</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="co">## You can be fancy and assign labels</span>
<span class="va">paramsDF</span><span class="op">$</span><span class="va">test_cat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span>
   <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">paramsDF</span><span class="op">$</span><span class="va">test_cat</span> <span class="op">%in%</span> <span class="st">"1"</span>,
      <span class="st">"up-regulated"</span>,
      <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">paramsDF</span><span class="op">$</span><span class="va">test_cat</span> <span class="op">%in%</span> <span class="st">"-1"</span>,
         <span class="st">"down-regulated"</span>,
         <span class="st">"unchanged"</span><span class="op">)</span><span class="op">)</span>,
   levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"up-regulated"</span>,
      <span class="st">"unchanged"</span>,
      <span class="st">"down-regulated"</span><span class="op">)</span><span class="op">)</span>;
<span class="va">gg_pre2</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/plot_ranged_enrichment.html">plot_ranged_enrichment</a></span><span class="op">(</span><span class="va">paramsDF</span>,
   test_cat_title<span class="op">=</span><span class="st">"RNAseq\ndirection"</span>,
   draw_points<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
   <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span>
<span class="va">gg_pre2</span></pre></div>
<p><img src="reference/figures/README-fancy_labels-1.png"><!-- --></p>
<p>If you’re curious, you can view the intermediate counts by using the helper function <code><a href="reference/plot_ranged_counts.html">plot_ranged_counts()</a></code>. One key feature of this plot is the distance at which the “non-overlapping” profiles recede down to zero. Eventually, at some infinite distance, every feature should overlap every peak, therefore the non-overlapping features should become zero.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="va">gg_prc</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/plot_ranged_counts.html">plot_ranged_counts</a></span><span class="op">(</span><span class="va">range_counts</span><span class="op">)</span> <span class="op">+</span>
   <span class="fu">facet_grid</span><span class="op">(</span><span class="va">Subclass</span><span class="op">~</span><span class="va">Experiment</span><span class="op">+</span><span class="va">PeakSet</span><span class="op">)</span>
<span class="va">gg_prc</span></pre></div>
<p><img src="reference/figures/README-plot_range_counts-1.png"><!-- --></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="http://github.com/jmw86069/peakdistanceenrichment/">http://​github.com/​jmw86069/​peakdistanceenrichment/​</a>
</li>
<li>Report a bug at <br><a href="http://github.com/jmw86069/peakdistanceenrichment/issues">http://​github.com/​jmw86069/​peakdistanceenrichment/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>James M. Ward <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-9510-2848" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by James M. Ward.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
