<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Peak-Distance-Enrichment • peakdistanceenrichment</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script>




<meta property="og:title" content="Peak-Distance-Enrichment" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-title-body">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">peakdistanceenrichment</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="http://github.com/jmw86069/peakdistanceenrichment/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Peak-Distance-Enrichment</h1>
    </div>

<div id="peak-distance-enrichment" class="section level1">

<p>This protocol describes the process for running distance-based enrichment methods used in Langer et al 2019, as shown in a visual workflow in Figure 2, supplement C.</p>
<ul>
<li>
<p>Provide “reference sites” that have genome coordinates, and a category assigned to each site.</p>
<ul>
<li>In this case the reference sites are transcript start sites (TSS) of genes tested by RNA-seq.</li>
<li>The category column named <code>"hit"</code> contains <code>-1</code> for down-regulated genes, <code>1</code> for up-regulated genes, and <code>0</code> for genes not changed.</li>
<li>Note that “reference sites” can have any category values relevant to the experiment, but must contain at least two values in order to test enrichment of one category compared to the other. In general, “reference sites” should represent the universe of sites detectable by the appropriate experimental protocol.</li>
</ul>
</li>
<li>
<p>Define “peaks” that will be tested for association to reference sites, where each peak has genome coordinates. Note for this workflow, strand is not considered.</p>
<ul>
<li>In this case there are two sets of peaks: ATAC-seq peaks that have <strong>higher</strong> chromatin accessibility upon BAF47 knockdown; and ATAC-seq peaks that have <strong>lower</strong> chromatin accessibility upon BAF47 knockdown.</li>
<li>In reality any set of peaks can be used, and multiple sets can be tested in series.</li>
</ul>
</li>
<li>
<p>Start at distance 0 and iterate through the following steps. Between each iteration, increase the distance by 5000 and repeat.</p>
<ul>
<li><p>Expand the set of “peaks” by the distance from the center of each peak. After this expansion, all “peaks” will be the same width.</p></li>
<li><p>Merge any overlapping peaks so that there are no overlapping regions. This step is sometimes called “merge”, “disjoin”, or “reduce”.</p></li>
<li>
<p>Tabulate the number of reference sites that overlap the expanded peak regions, for each category value; also tabulate the number of reference sites that do not overlap the expanded peak regions, for each category value.</p>
<ul>
<li>Using the example data, this step will produce 6 values:</li>
</ul>
<ol>
<li>Number of sites with category <code>-1</code> that overlap peaks</li>
<li>Number of sites with category <code>0</code> that overlap peaks</li>
<li>Number of sites with category <code>1</code> that overlap peaks</li>
<li>Number of sites with category <code>-1</code> that do not overlap peaks</li>
<li>Number of sites with category <code>0</code> that do not overlap peaks</li>
<li>Number of sites with category <code>1</code> that do not overlap peaks</li>
</ol>
</li>
</ul>
</li>
<li>
<p>For each distance threshold, an enrichment test is performed using one or more “test category values”, compared to the other “background category values”.</p>
<ul>
<li>For the example above, one may test <code>1</code> for enrichment compared to <code>0, -1</code>.</li>
<li>Alternatively, one may test <code>-1, 1</code> for enrichment compared to <code>0</code>.</li>
</ul>
</li>
<li>
<p>A hypergeometric enrichment test is performed, which requires the following four values:</p>
<ol>
<li>Number of sites that overlap peaks, which have “test category values”.</li>
<li>Number of sites that overlap peaks, which have “background category values”.</li>
<li>Number of sites that do not overlap peaks, which have “test category values”.</li>
<li>Number of sites that do not overlap peaks, which have “background category values”.</li>
</ol>
</li>
<li>
<p>In R, the function <code><a href="https://rdrr.io/r/stats/Hypergeometric.html">phyper()</a></code> is used to perform this test, with the following arguments:</p>
<ul>
<li>
<code>q</code> - number of sites with category of interest that overlap peaks <code>(1)</code>
</li>
<li>
<code>m</code> - total number of sites with category of interest <code>(1 + 3)</code>
</li>
<li>
<code>n</code> - total number of sites that do not have the category of interest <code>(2 + 4)</code>
</li>
<li>
<code>k</code> - number of sites that overlap peaks <code>(1 + 2)</code>
</li>
</ul>
</li>
</ul>
<div id="r-analysis-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#r-analysis-workflow" class="anchor"></a>R analysis workflow</h2>
<p>There is an R package in a Github repository: <a href="https://github.com/jmw86069/peakdistanceenrichment" class="uri">https://github.com/jmw86069/peakdistanceenrichment</a></p>
<p>The R package can be installed by opening an R session and running this command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/remotes/man/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"jmw86069/peakdistanceenrichment"</span><span class="op">)</span></pre></div>
<p>The package installs four example files, sufficient to reproduce Figure 2B of Langer et al 2019, eLife.</p>
<ul>
<li>`hg19.sizes.txt - chromosome sizes</li>
<li>
<code>BAF47KD_steadystate_RNAseq_hits.bed</code> - reference TSS sites with <code>-1,0,1</code> hits</li>
<li>
<code>BAF47KD_steadystate_higher_accessibility_peaks.bed</code> - test peaks</li>
<li>
<code>BAF47KD_steadystate_lower_accessibility_peaks.bed</code> - test peaks</li>
</ul>
<p>See the README.Rmd to follow steps that will reproduce the figure.</p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>

</div>



      <footer>
      <div class="copyright">
  <p>Developed by James M. Ward.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


